lib/rules/indent-legacy.js

```diff
@@ -696,20 +696,6 @@ module.exports = {
             return startLine === endLine;
         }
 
-        /**
-         * Check to see if the first element inside an array is an object and on the same line as the node
```

Comment:
Do you mind explaining why we need this change?

Reply:
The function is now `no-unused-vars` error.

An alternative is to leave it in the code with `eslint-disable-*`, or to use `eslint-disable-*` on the removed `else if` line.

Reply:
Sorry, should have been clearer (I was referring to the full change). Why do we need to remove the `else if` below?

Reply:
It's an error reported by `no-dupe-else-if` (which is now enabled in the codebase via `eslint:recommended`) and because of the linting error `npm test` fails.

```js
if (!parentVarNode || ...) {
    // ...
} else if (!parentVarNode && ...) {
   nodeIndent += indentSize;
}
```

The whole condition in the `else if` can be `true` only when `!parentVarNode` is `true`, but in that case the condition in the first `if` will be `true` and it will not proceed to `else`, meaning that `nodeIndent += indentSize;` is a dead code.

Reply:
Understood, thanks!